src/data/helper.rs

```diff
@@ -30,3 +31,32 @@ pub fn build_year(
 
     map.insert(year, m);
 }
+
+#[allow(clippy::needless_pass_by_value)]
+pub fn build_subdivision_year(
+    years: Option<&std::ops::Range<Year>>,
+    year: Year,
+    national_holidays: &mut HolidayPerCountryMap,
+    subdivision_holidays: impl IntoIterator<Item = (NaiveDate, &'static str)>,
+    map: &mut HolidayPerCountryMap,
+    country: Country,
+    county_name: &(impl ToString + ?Sized),
+) {
+    if !should_build_year(years, year) {
+        return;
+    }
+
+    let m = national_holidays
```

Comment:
I'm not sure if it is possible to have a national and a subdivision holiday on the same day, but they could be merged into one with this code instead:

```rs
    let mut m = national_holidays.remove(&year).unwrap_or_default();

    for (date, name) in subdivision_holidays {
        let national_holiday_name = m.remove(&date).map(|h| h.name + ", ").unwrap_or_default();

        m.insert(
            date,
            Holiday::new(country, national_holiday_name + county_name, date, name),
        );
    }
```

Reply:
I guess this approach is still using big memory size. 

Your approach is merge national holiday + subidivsion holiday at "build" time. My suggestion is to do the merge at runtime, i.e. "get", "iter" or "contains". Could you consider this approach?

Reply:
Sorry it took so long. Let me know if you are happy with my implementation